yogstation/code/modules/power/nuclear_engine.dm

```diff
@@ -0,0 +1,449 @@
+	var/AmtAdd
+	for(var/obj/item/twohanded/required/FuelRod/FR in FRS)
+		if(!istype(FR,/obj/item/twohanded/required/FuelRod))
+			return
+		AmtAdd += FR.FuelPower
+	var/final = AmtAdd
+	return final //IE, AmtSubtract with one CR = 1, AmtAdd with 2 Uranium = 4, so resultant reaction power is 3, which also means a heat spike
+
+/datum/nuclearreaction/proc/ConsumeFuel()
+	for(var/obj/item/twohanded/required/FuelRod/FR in FRS)
+		if(FR.loc != reactor || !FR in reactor.contents)
+			FRS -= FR
+		if(!istype(FR,/obj/item/twohanded/required/FuelRod))
+			return
+		if(FR.integrity <= 0) //That rod's spent, make it into DU
+			reactor.say("[FR] has been depleted")
+			DepletedAmt ++
+			qdel(FR)
+			FRS -= FR
+			FR = null
+			var/obj/item/twohanded/required/FuelRod/depleted/spent = new
+			FRS += spent
+			spent.forceMove(reactor)
+			return
+		if(!istype(FR, /obj/item/twohanded/required/FuelRod/depleted)) //Depleted rods don't deplete further, they just interrupt heat
+			FR.integrity -= 10 - reactor.Efficiency/10 // Eg, 100/10 = 10, so integrity -= 0 aka no health lost, 25 = 2.5, so 8.5 health lost, so keep that efficiency up!
+	for(var/obj/item/twohanded/required/ControlRod/CR in CRS)
+		if(CR.loc != reactor || !CR in reactor.contents)
+			CRS -= CR
+		if(!istype(CR,/obj/item/twohanded/required/ControlRod))
+			return
+		if(CR.integrity <= 10) //CR is spent!
+			qdel(CR)
+			reactor.say("[CR] has worn out.")
+		CR.integrity -= 10 - reactor.Efficiency/10
+
+/obj/item/twohanded/required/ControlRod
+	name = "Carbon Control Rod"
+	desc = "A huge rod of carbon used for nuclear reactors. It is inert."
+	icon = 'icons/obj/control_rod.dmi'
+	icon_state = "carbonrod"
+	var/integrity = 10000 //As rods become worn, they are damaged.
+	max_integrity = 10000
+	var/obj/machinery/power/NuclearReactor/reactor
+	var/ControlPower = 4.8 // 1 Rod of this will cockblock 1 fuel rod
+
+/obj/item/twohanded/required/FuelRod
+	name = "Uranium Fuel Rod"
+	desc = "A huge stick of uranium to be inserted into a nuclear reactor, its casing is in tact, mitigating most of its radioactive output"
+	icon = 'icons/obj/control_rod.dmi'
+	icon_state = "uraniumrod"
+	var/integrity = 10000//As rods become worn, they are damaged.
+	max_integrity = 10000
+	var/obj/machinery/power/NuclearReactor/reactor
+	var/FuelPower = 5 //How strong is this fuel rod? higher = more heat produced
+	var/radiation_strength = 10
+	var/FuelPenalty = 0
+
+/obj/item/twohanded/required/FuelRod/attack_hand(mob/user)
+	radiation_pulse(src,radiation_strength, 2)
+	. = ..()
+
+/obj/item/twohanded/required/FuelRod/plasma
+	name = "Plasma Fuel Rod"
+	desc = "A huge stick of plasma to be inserted into a nuclear reactor. It has a small label haphazardly stuck to it: WARNING, UNCONTROLLED USE OF THIS WILL OVERHEAT REACTOR. USE AT OWN RISK!" //This isn't a question of if it fucks your setup... It's when.
+	icon_state = "plasmarod"
+	integrity = 1000//As rods become worn, they are damaged.
```

Comment:
you don't have to set integrity and max integrity here since they are inherited from the uranium rod

Reply:
Intentional, it's 1K instead of 10K

Reply:
misread